src/components/Modal/CartModal.tsx

```diff
@@ -0,0 +1,98 @@
+import Modal from './Modal/Modal';
+import * as styles from './CartModal.style';
+import Button from '../Button/Button';
+import Counter from '../Counter/Counter';
+import Image from '../Image/Image';
+import { RemoveFromCartButton } from '../CartButton/CartButton';
+import { useApiContext } from '../../contexts/ApiContext';
+import { useErrorContext } from '../../contexts/ErrorContext';
+import getCartItems from '../../api/getCartItems';
+import patchCartItem from '../../api/patchCartItem';
+import { deleteCartItem } from '../../api/deleteCartItem';
+import { createCartItemViewModel } from '../../api/model/createCartItemViewModel';
+
+interface CartModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export default function CartModal({ isOpen, onClose }: CartModalProps) {
+  const { showError } = useErrorContext();
+  const { data: cartItems, fetcher: refetchCart } = useApiContext({ fetchFn: getCartItems, key: 'getCartItems' });
```

Comment:
cartItems 를 다시 조회하기 위해 동일한 코드가 여러번 호출되고 있는 것으로 보이는데요. 매번 같은 key 와 fetchFn 을 넘겨주며 사용하고 있으므로 이 로직 자체를 별도의 훅으로 빼는 것을 제안드리고 싶습니다.

```suggestion
  const { data: cartItems, fetcher: refetchCart } = useCartItemsApi();
```

이렇게 할 경우 다음의 효과를 누릴 수 있을 것이라 생각합니다.

- 작성해야 하는 중복 코드 수 감소
- key 를 의도치 않게 잘못 설정하여 서로 다른 데이터를 참조하는 오류 가능성 감소 (SSOT)

메타는 어떻게 생각하시는지요?

Reply:
[refactor: useCartItems 중복 코드 줄이기](https://github.com/woowacourse/react-shopping-products/pull/148/commits/6a3165d26a22057ea3f258568d50ef63133c27bd)
합리적이라고 생각합니다! 키 값을 잘못 입력할 수 있는 위험도 없앨 수 있다고 생각해요 ㅎㅎ